packages/fast-colors/src/color-quantization.ts

```diff
@@ -0,0 +1,197 @@
+// tslint:disable:member-ordering
+// tslint:disable:no-bitwise
+// tslint:disable:prefer-for-of
+
+import { PixelBlob } from "./pixel-blob";
+import { Histogram } from "./histogram";
+import { insertIntoSortedList, PixelBox } from "./pixel-box";
+import { ColorRGBA64 } from "./color-rgba-64";
+
+export interface QuantizedColor {
+    color: ColorRGBA64;
+    pixelCount: number;
+    colorVolume: number;
+}
+
+export interface QuantizeConfig {
+    // must be in the range [1,8]. Memory use increases as 4*2^(3*signifigantBits).
+    // setting signifigantBits to 8 requires a 64 megabyte histogram
+    signifigantBits: number;
+    // lowering this value increases the CPU load but includes more pixels in the calculation
+    pixelSkipping: number;
+    // desired output palette size. the actual output may vary in edge cases such as images with very few colors.
+    targetPaletteSize: number;
+    // For a final palette of size targetPaletteSize, we determine the first fractionByPopulation*targetPaletteSize
+    // using population as the only factor when determening sort order. For the rest of the colors the
+    // sort order is population * colorVolume. This helps highly contrasting colors in a small area to show
+    // up in some of the final output.
+    fractionByPopulation: number;
+    // This predicate can be used to screen out undesirable colors from the final output
+    // eg: excluding colors with a pixelCount below a min value
+    isBoxValid: ((box: PixelBox) => boolean) | null;
+    // This predicate can be used to exlude pixels from the histogram.
+    // It is passed numbers in the range [0,255] in rgba order
+    // EG: excluding colors too close to pure white
+    isHistoPixelValid: ((pixel: number[]) => boolean) | null;
```

Comment:
Minor naming nit - we try to use the full word when there isn't really a standardized abbreviation - can we make this `isHistogramPixelValid`?